lib/vmpooler/api/v1.rb

```diff
@@ -1063,6 +1074,37 @@ def invalid_pool(payload)
       JSON.pretty_generate(result)
     end
 
+    post "#{api_prefix}/poolreset/?" do
+      content_type :json
+      result = { 'ok' => false }
+
+      if config['experimental_features']
+        need_token! if Vmpooler::API.settings.config[:auth]
+
+        payload = JSON.parse(request.body.read)
```

Comment:
Since this payload isn't `rescue`d, a user will get a mysterious 500 error if they post bad JSON.

Reply:
It looks like this is something that we do throughout vmpooler and will affect other POST methods in the vmpooler API. Should we try to rescue this and give a 500 error instead?

Reply:
Rescue it, I'd set a `400` (Bad request) with a message like `JSON payload could not be parsed`. Just something easy so that an API user knows there is something they can do to fix their request and that the server isn't just angry. 😄 

As for the other API endpoints, ideally I'd like if all of them had better error handling and user-friendly error messages but I figure we should provide some better messaging as we add new endpoints.

Reply:
Thanks. I have updated this so poolreset will give a 400 and a message if bad JSON is provided. I tested it and validated I get the 400 and message about JSON not being able to be parsed when bad JSON is provided.